privacyidea/lib/tokens/pushtoken.py

```diff
@@ -189,6 +189,33 @@ def _build_smartphone_data(serial, challenge, registration_url, pem_privkey, opt
     message_on_mobile = get_action_values_from_options(SCOPE.AUTH,
                                                        PUSH_ACTION.MOBILE_TEXT,
                                                        options) or DEFAULT_MOBILE_TEXT
+    # Get the request object
+    req_headers = options.get("g", {}).get("request_headers")
+    request = None
+    if req_headers:
+        req_environment = req_headers.environ
+        request = req_environment.get("werkzeug.request")
+    if request:
+        user_object = request.User
+    else:
+        # Get owner from token object
+        try:
+            tok = get_one_token(serial=serial)
```

Comment:
To avoid another db lookup, we could just pass the token object itself to the function and get serial and assigned user from it.

Reply:
Will check and do.

Reply:
Done in 6c5aed4c